### ISSUE TITLE
TDeintMod broke with "Property read unsuccessful on map with error set but no error output" gobbledygook

### ISSUE THREAD

**Issue by v-fox** [vapoursynth/vapoursynth#955]
[openSUSE recently updated](https://build.opensuse.org/package/show/home:X0F:HSF/vapoursynth) its ancient version of VS to the current one which caused [TDeintMod](https://github.com/HomeOfVapourSynthEvolution/VapourSynth-TDeintMod#example-usage-of-iscombed)-based pre-interpolation [deinterlacing stage](https://gist.github.com/v-fox/43c287426c366679afc4c65eece60cbc#file-motioninterpolation-vpy-L992) of my [mvtools](https://github.com/dubhater/vapoursynth-mvtools)-based motion interpolation script to fail with this incomprehensible error, followed by whole [mpv](https://github.com/mpv-player/mpv) segfault: `Property read unsuccessful on map with error set but no error output: clip`.

Please advise on how to work around it or to replace TDeintMod with something maintained to make sure that old crappy DVD videos are automatically safely deinterlaced before generating motion vectors.

**Comment by Setsugennoao** [comment]
Duplicate of https://github.com/vapoursynth/vapoursynth/issues/950

**Comment by v-fox** [comment]
This is completely different error and I have no idea what you mean by "the issue is not VapourSynth related, but rather a (very bad) config issue". It's a script that worked fine for years with crashing part being copied from the plugin's manual, there is no 'configs' for it. How is any of that relevant?

**Comment by Setsugennoao** [comment]
You were on API3, now API4. You should compile API3 plugins with API3 headers but it somehow isn't the case for you.

Anyways, you can use vivtc to detect the combing.

**Comment by v-fox** [comment]
> You were on API3, now API4. You should compile API3 plugins with API3 headers but it somehow isn't the case for you.

All packages for openSUSE are compiled by Open Build Service in VMs using each other. Meaning that VS is compiled against latest system packages and then all plugins are compiled against it. There is no compilation errors for TDeintMod, despite it being long unmaintained.
So, there is no indication for any of that and I have no clue how to "compile API3 plugins with API3 headers" other than keeping another package for ancient version of VS. But then how that supposed to load into main VS version?

The error itself is sudden and meaningless. And by your link without obvious self-evidence and explanation I assume that you're implying that API has changed to break compatibility with no regard for backward-compatibility with abandoned plugins - a typical "defective by design"-scenario which isn't a right thing to do, to say the least.

TDeintMod and MVTools both are saying `VAPOURSYNTH_API_VERSION, 1` in their code, which is neither 3 or 4, and are using `clip:clip`, yet MVTools works while TDeintMod does this. Changing `clip:clip` to `clip:vnode` (as in #950) in TDeintMod/TDeintMod.cpp compiles but makes it be silently ignored by VS altogether. At least with that it errors out gracefully instead of crashing the whole player.
I guess maybe it actually doesn't like that error-related part in: https://github.com/HomeOfVapourSynthEvolution/VapourSynth-TDeintMod/blob/044e057b5fcc2f880441f6fce1da1e66cf5f7303/TDeintMod/TDeintMod.cpp#L2133

> Anyways, you can use vivtc to detect the combing.

Thanks, I'll look into it. But at [a glance](https://github.com/vapoursynth/vivtc/blob/master/docs/vivtc.rst) I already see an issue: it seems to be limited to 8-bit input while tdeint seems to be eating up anything up to 16 meaning that for the script to transparently run on modern 10/12-bit h264/h265 videos it would need some additional testing logic. But, I guess, these videos should not be in interlacing-detection stage anyway.

**Comment by Setsugennoao** [comment]
The thing is, if you just compile an API3 plugin with `VAPOURSYNTH_API_MAJOR=4`, you will get 
![image](https://github.com/vapoursynth/vapoursynth/assets/41454651/10ab0fde-5455-4fcc-aa0b-8ecccd115c12)

As it checks for the correct versioning [here](https://github.com/vapoursynth/vapoursynth/blob/master/src/core/vscore.cpp#L438).

----------

> TDeintMod and MVTools both are saying VAPOURSYNTH_API_VERSION, 1 in their code, which is neither 3 or 4

The `VAPOURSYNTH_API_VERSION` constant changes based on the header used. [v3](https://github.com/vapoursynth/vapoursynth/blob/master/include/VapourSynth.h#L26-L28), [v4](https://github.com/vapoursynth/vapoursynth/blob/master/include/VapourSynth4.h#L28-L30).
1 has nothing to do with versioning, it's just true value for marking the plugin as readonly [here](https://github.com/vapoursynth/vapoursynth/blob/master/include/VapourSynth.h#L237).


So, in conclusion: I was wrong at the start, since to even load the plugin you need to have the correct header to define `VAPOURSYNTH_API_VERSION` with `API_MAJOR=3`, so it doesn't crash on api misuse. It's something more fucked up I can't reproduce on my machine, but is definitely not related to normal workflow.


----------

> But at a glance I already see an issue: it seems to be limited to 8-bit input while tdeint ...

You can do something like:
```py
src = ...  # YUV422P10

vfM_in = src.resize.Bicubic(format=vs.YUV420P8)
vfM = vfM_in.vivtc.VFM(..., clip2=src)
```

**Comment by Setsugennoao** [comment]
Ok, the error is thrown here: <https://github.com/HomeOfVapourSynthEvolution/VapourSynth-TDeintMod/blob/master/TDeintMod/TDeintMod.cpp#L2058>.

You somehow are passing an `in` map (aka from the invoke call in the cython package) with an error.
<https://github.com/vapoursynth/vapoursynth/blob/master/src/cython/vapoursynth.pyx#L2817-L2839>

**Comment by Setsugennoao** [comment]
I also checked mpv code, no errors that could happen with the APIv4 transition.
Will have to let Myrsloik debug this one.

By the way, a couple of things in your script are completely broken:
https://gist.github.com/v-fox/43c287426c366679afc4c65eece60cbc#file-motioninterpolation-vpy-L981
https://gist.github.com/v-fox/43c287426c366679afc4c65eece60cbc#file-motioninterpolation-vpy-L1003

**Comment by myrsloik** [comment]
Fixed in master
